### PR TITLE
Fixing the lost changes for Marking Desks

### PR DESCRIPTION
**Pull request by akintolga**
- Some of the changes for the original PR is lost, this is to recover some of the lost staff

### CODE REVIEW

**Comment by akintolga** [comment]
@mdhaman @nrvikas

File: scripts/apps/authoring/authoring/directives/AuthoringDirective.js [hunk 1]
```diff
@@ -846,12 +846,18 @@ export function AuthoringDirective(superdesk, superdeskFlags, authoringWorkspace
 
             function refreshMarkedItems(field, data) {
                 if ($scope.item._id === data.item_id) {
+                    let mark = data.mark_id;
```
Comment (gbbr) [inline scripts/apps/authoring/authoring/directives/AuthoringDirective.js]: Might be cleaner to do:
```js
let mark = field === 'marked_desks' ? {desk_id: data.mark_id} : data.mark_id;
```

File: scripts/apps/archive/directives/MarkedItemTitle.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+            scope.toggleClass = function(isOpen) {
+                scope.open = isOpen;
+            };
+
+            scope.marks = scope.item[scope.markField];
+            scope.hasMarkItemPrivilege = authoring.itemActions(scope.item).mark_item_for_desks;
+
+            scope.$on('item:' + scope.markField, ($event, data) => {
+                let marks = scope.marks || [];
+                let mark = data.mark_id;
```
Comment (gbbr) [inline scripts/apps/archive/directives/MarkedItemTitle.js]: You could also simplify here by doing:
```js
let mark = scope.markField !== 'marked_desks' ? data.mark_id : {desk_id: data.mark_id};
```

File: scripts/apps/archive/directives/MarkedItemTitle.js [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+                if (scope.markField === 'marked_desks') {
+                    mark = {desk_id: data.mark_id};
+                }
+
+                if (scope.item._id === data.item_id) {
```
Comment (gbbr) [inline scripts/apps/archive/directives/MarkedItemTitle.js]: If this condition is false, this even listener takes no action, so instead, you could again simplify and reduce indentation by placing this at the top of the function (on line 35):
```js
if (scope.item._id !== data.item_id) {
    return;
}
```
And then continue on without any more `if`'s

File: scripts/apps/archive/directives/MarkedItemTitle.js [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+                        } else {
+                            _.remove(marks, _.filter(marks, mark)[0]);
+                            scope.marks = marks;
+                        }
+                    });
+                }
+            });
+
+            // Check if the directive is created to mark desks or highlights
+            if (scope.markField === 'marked_desks') {
```
Comment (gbbr) [inline scripts/apps/archive/directives/MarkedItemTitle.js]: You have already written this `if` statement above. Why not include the body of it there? We don't have to have the same `if` statement twice in the same block. Could we merge instead?

File: scripts/apps/archive/directives/MarkedItemTitle.js [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+                scope.service = highlightsService;
+                scope.fetchFunction = 'get';
+                scope.className = 'icon-star red';
+                if (scope.marks && scope.marks.length > 1) {
+                    scope.className = 'icon-multi-star red';
+                }
+            }
+
+            scope.$watch('item.' + scope.markField, (marks) => {
+                if (marks) {
```
Comment (gbbr) [inline scripts/apps/archive/directives/MarkedItemTitle.js]: Reduce indentation here by doing:
```js
if (!marks) {
    return;
}
```
instead.

File: scripts/apps/archive/directives/MarkedItemTitle.js [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+            var closeTimeout, popup;
+
+            elem.on({
+                click: function(e) {
+                    e.preventDefault();
+                    e.stopPropagation();
+                },
+                mouseenter: function(e) {
+                    popup = $(this).find('.highlights-list');
+                    popup.not('.open').children('.dropdown-toggle')
```
Comment (gbbr) [inline scripts/apps/archive/directives/MarkedItemTitle.js]:
```js
popup.not('.open')
    .children('.dropdown-toggle')
    .click();
```

**Comment by mdhaman** [comment]
closing this PR you review comments will be addressed in PR #1250

**Pull request closed without merging**